### PR TITLE
Expose output buffer config in the API

### PR DESCRIPTION
**Pull request by ryzhyk**
My earlier PR that introduced output buffering did not include a REST
API to configure it. This commit adds such an API.  It does so in a
somewhat roundabout way though.  The challenge is that output buffer
config is not strictly part of the connector cofig, rather it's
an attribute of the endpoint (i.e., "attached connector" in the pipeline
manager terminology).  In the UI, this corresponds to the dotted arrow
connecting the pipeline with the connector.  In reality, I expect that
users will want to configure output buffers as part of the connector
rather than deal with separate per-endpoint settings and even if we want
to add per-endpoint config in the future, the default buffering settings
should come from the connector config.

This commit introduces the `ConnectorConfigTemplate` type, which includes
`ConnectorConfig` and `OutputBufferConfig` fields and is used in the API
instead of `ConnectorConfig`.

@Karakatiza666 , this will require support in the WebConsole (#1591)

Is this a user-visible change (yes/no): yes

### CODE REVIEW

**Review by mihaibudiu — APPROVED** [review]

**Review by snkas — COMMENTED** [review]
I understand that the output buffer strategy is strictly speaking part of the endpoint (attached connector), rather than the connector itself. As such, the direct way would be to have it be part of AttachedConnector in pipeline-manager. As it now becomes part of the ConnectorConfig in any case (thus, not allowing per-attachment output buffering specification), what's the motivation to not have it be an optional field "output_buffer" next to the already "transport" and "format"?

**Comment by ryzhyk** [comment]
> I understand that the output buffer strategy is strictly speaking part of the endpoint (attached connector), rather than the connector itself. As such, the direct way would be to have it be part of AttachedConnector in pipeline-manager. As it now becomes part of the ConnectorConfig in any case (thus, not allowing per-attachment output buffering specification), what's the motivation to not have it be an optional field "output_buffer" next to the already "transport" and "format"?

I will do that. The main downside is that we don't have separate input and output connector configs, so this field will appear in input connector configs, where it doesn't have any meaning.

File: openapi.json [hunk 1]
```diff
@@ -841,14 +841,14 @@
...
-                      "enable_buffer": false,
+                      "enable_output_buffer": false,
                       "format": {
                         "config": null,
                         "name": "csv"
                       },
-                      "max_buffer_size_records": 0,
-                      "max_buffer_time_millis": 0,
-                      "max_buffered_records": 1000000,
+                      "max_output_buffer_size_records": 18446744073709551615,
```
[the remark this answers is not in the document]
Comment (ryzhyk) [reply openapi.json:849]: It didn't change.  `max_buffered_records` is now called `max_queued_records`; this one is a new setting related to the output buffer.  I will add some docs clarifying the distinction.

File: Earthfile [hunk 2]
```diff
@@ -528,6 +528,6 @@ all-tests:
     BUILD +test-docker-compose-stable
     BUILD +test-debezium-mysql
     BUILD +test-debezium-jdbc-sink
-    BUILD +test-snowflake
+    # BUILD +test-snowflake
```
[the remark this answers is not in the document]
Comment (ryzhyk) [reply Earthfile:531]: No, but it's currently broken because our snowflake account is disabled, so I had to disable it for CI to pass.

**Pull request merged**